sources/packages/backend/apps/api/src/services/application-bulk-withdrawal/application-bulk-withdrawal-import-business-validation.models.ts

```diff
@@ -0,0 +1,248 @@
+ * the offering from being saved.
+ */
+export enum ValidationContextTypes {
+  Warning = "warning",
+}
+
+/**
+ * Represent the context of an error that has an additional context to
+ * express a possible warning or some relevant information captured during
+ * the validation process.
+ * All validations when failed will generate an error. The ones that have
+ * the warning or information contexts will not be considered critical.
+ */
+export class ValidationContext {
+  /**
+   * Creates an error context that will make the error downgrade to
+   * a condition of a warning information will force the offering
+   * to be reviewed by the Ministry.
+   * @param warningTypeCode warning code that uniquely identifies this condition.
+   * @returns the warning context.
+   */
+  static CreateWarning(
+    warningTypeCode: ApplicationWithdrawalValidationWarnings,
+  ): ValidationContext {
+    const newContext = new ValidationContext();
+    newContext.type = ValidationContextTypes.Warning;
+    newContext.typeCode = warningTypeCode;
+    return newContext;
+  }
+
+  /**
+   * Context type.
+   */
+  type: ValidationContextTypes;
+  /**
+   * Unique identifier of the validation error.
+   */
+  typeCode: ApplicationWithdrawalValidationWarnings;
+}
+
+export class ApplicationBulkWithdrawalImportBusinessValidationModel {
+  /**
+   * SIN.
+   */
+  @IsValidSIN({
+    message: `${DataTextHeaders.sin} must be a valid SIN.`,
+  })
+  @Length(SIN_NUMBER_LENGTH, SIN_NUMBER_LENGTH, {
+    message: `${DataTextHeaders.sin} must be of length ${SIN_NUMBER_LENGTH}.`,
+  })
+  sin: string;
+  /**
+   * Application Number.
+   */
+  @IsNumberString(
+    { no_symbols: true },
+    {
+      message: `${DataTextHeaders.applicationNumber} must be a valid number.`,
+    },
+  )
+  @Length(APPLICATION_NUMBER_LENGTH, APPLICATION_NUMBER_LENGTH, {
+    message: `${DataTextHeaders.applicationNumber} must be of length ${APPLICATION_NUMBER_LENGTH}.`,
+  })
+  applicationNumber: string;
+  /**
+   * Withdrawal Date.
+   */
+  @IsDate({
+    message: `${DataTextHeaders.withdrawalDate} must be a valid withdrawal date.`,
+  })
+  withdrawalDate: Date;
+  /**
+   * Application found.
+   */
+  @IsIn([true], {
+    message:
+      "The application is not present in SIMS and is a part of the SFAS system.",
+    context: ValidationContext.CreateWarning(
+      ApplicationWithdrawalValidationWarnings.ApplicationNotFound,
+    ),
+  })
+  applicationFound: boolean;
+  /**
+   * Application belongs to institution.
+   */
+  @ValidateIf(
+    (object: ApplicationBulkWithdrawalImportBusinessValidationModel) =>
+      object.applicationFound,
+  )
+  @IsIn([true], {
+    message: "The application does not belong to this institution.",
+  })
+  applicationBelongsToInstitution?: boolean;
+  /**
+   * Valid SIN.
+   */
+  @ValidateIf(
+    (object: ApplicationBulkWithdrawalImportBusinessValidationModel) =>
+      object.applicationFound,
+  )
+  studentSINMatch?: boolean;
```

Comment:
`applicationFound` and this `studentSINMatch` has same effect as the the AC. Why is this not treated as warning?